[[https://melpa.org/#/sequed][file:https://melpa.org/packages/sequed-badge.svg]]
[[./sequed-aln-mode.png]]

* sequed
 
An major mode for editing, viewing and transforming DNA sequence alignments in GNU Emacs.

** Overview 

The =sequed= package provides some basic functionality for editing, viewing and manipulating DNA sequence data in Emacs. In =sequed-mode= commands are available for navigating and editing DNA sequences in fasta file format, as well as other common Bioinformatics tasks such as creating reverse-complements, translating selected regions into amino acid sequences, etc. In =sequed-aln-mode= commands are available for pretty viewing (and navigating) of fasta format alignments of multiple sequences, possibly reducing the view to specific subregions. To use =sequed= you should be familiar with DNA biology and basic Emacs commands.       

** Installation

=sequed= is available through [[http://melpa.milkbox.net/][melpa]] making installation using the Emacs package manager very simple.
First, add melpa to your list of package archives by following these [[https://melpa.org/#/getting-started][melpa how-to-install instructions]]. Once this is done the installation is trivial:

#+BEGIN_EXAMPLE
    M-x package-install sequed
#+END_EXAMPLE

Then add =(require 'sequed)= to your =~/.emacs= or =~/.emacs.d/init.el=. If you use the =use-package= system for managing your
Emacs packages then simply add the following to your =.emacs= or =~/.emacs.d/init.el=:

#+BEGIN_EXAMPLE
(use-package sequed
  :ensure t)
#+END_EXAMPLE

** Quickstart
After installing sequed and restarting Emacs, sequed-mode automatically becomes the major mode for files with .fa or .aln endings. You can manually invoke the sequed-mode major mode with

#+BEGIN_EXAMPLE
    M-x sequed-mode
#+END_EXAMPLE

** Manipulating DNA Sequences in Sequed Mode

*** Moving around and selecting text

*** Sequence transformations

*** Exporting sequences in other formats

** Viewing DNA Alignments in Sequed-Aln Mode 
When visiting a buffer in =sequed-mode= that contains a valid sequence alignment you
can create a new read-only buffer containing a colorful alignment view using either the 
key combination =C-c C-a= or =M-x sequed-mkaln=. You will be prompted for the first position (base)
and last position (base) of the region to be displayed. Emacs can be slow displaying files
with long lines so it is recommended that you view regions of about 1kb or less at one time.
It is quick and easy to open a new display to look at other regions of a long sequence
using the commands outline above. In =sequed-aln-mode= the absolute position of the current
base (cursor position) and the name of the sequence that the cursor is positioned in are
both displayed in the mode line at the bottom of the screen. The command =C-c C-b= is used to specify a base position (absolute position in the orginal sequences) to move the cursor to. So, if you are
viewing the region from 1500 to 2000 valid arguments for base position are between 1500 and 2000.
You will receive a "base outside of region" error message if in this case you specifed for example position 500.

The new buffer will have major mode =sequed-aln-mode==.
